python/cmsis_svd/model.py

```diff
@@ -147,9 +148,11 @@ def get_derived_from(self):
         if self.derived_from is None:
             return None
 
-        for field in self.parent.fields:
-            if field.name == self.derived_from:
-                return field
+        f = [f for f in self.parent.fields if f.name == self.derived_from][0]
+        if(f.get_derived_from() == None):
+            return f
+        else:
+            return f.get_derived_from()
```

Comment:
As is, this will raise a different exception in the case that would have previously raised a KeyError.  I suggest the following as an alternative that maintains the existing behavior.  This also avoid multiple calls to get_derived_from() which might recurse on that path (not expecting that performance would be worth concerning ourself with too much but it exists):

```python
        for field in self.parent.fields:
            if field.name == self.derived_from:
                next_level_derive = field.get_derived_from()
                return next_level_derive if next_level_derive else field
```

Reply:
hello paul, as i just ran it on the whole corpus of data as-is and it doesn't raise any exception for me, on which file do you have that ?